parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,218 @@
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<const ParseTreeNode*> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<const ParseTreeNode*>> *container_child_fields) const override {
+
+    inline_field_names->push_back("property");
+    inline_field_values->push_back(getPropertyString());
+
+    inline_field_names->push_back("value");
+
+    // pretty print multiple values as a list
+    if(values_->size() > 1) {
+      std::string str("(");
+      for(const ParseString& parseString : *values_) {
+        str += parseString.value() + ',';
+      }
+      str.erase(str.size() - 1, 1);
+      str += ")";
+      inline_field_values->push_back(str);
+    } else {
+      inline_field_values->push_back(values_->begin()->value());
+    }
+  }
+
+ private:
+
+  Property property_;
+  std::unique_ptr<PtrList<ParseString>> values_;
+
+  DISALLOW_COPY_AND_ASSIGN(ParseBlockPropertyItem);
+};
+
+/**
+ * @brief Contains user-specified physical properties of a table's blocks as BlockPropertyItem's
+ */
+class ParseBlockProperties : public ParseTreeNode {
+ public:
+
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   **/
+  ParseBlockProperties( int line_number, 
+                        int column_number, 
+                        PtrList<ParseBlockPropertyItem>* properties)
+                                : ParseTreeNode(line_number, column_number),
+                                  properties_(properties) { }
+
+  std::string getName() const override {
+    return "BlockProperties";
+  }
+
+  /**
+   * @brief Gets the specified BlockPropertyItem.
+   * 
+   * @param property The property, one of kSort, kCompress, or kType.
+   * @return Pointer to the property or nullptr if not found.
+   */
+  const ParseBlockPropertyItem* getPropertyItem(ParseBlockPropertyItem::Property property) const {
+    for(auto propertyItem = properties_->begin(); propertyItem != properties_->end(); ++propertyItem) {
+      if(propertyItem->property() == property) {
+        return &*propertyItem;
+      }
+    }
+    return nullptr;
+  }
+
+ protected:
+  void getFieldStringItems(
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<const ParseTreeNode*> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<const ParseTreeNode*>> *container_child_fields) const override {
+
+    for(auto propertyItem = properties_->begin(); propertyItem != properties_->end(); ++propertyItem) {
```

Comment:
Really, a `ParseBlockProperties` object is a container of `ParseBlockPropertyItem`s. So, you would have a single container child field name "properties" and all the contained properties as a single entry in the fields underneath it.